### PR TITLE
java tls cleanup

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Cleans and re-order java tls program code base.

### Motivation
The code was messy and mixed global variables and struct members.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/ebpf_javatls.go [hunk 1]
```diff
@@ -36,195 +36,176 @@ const (
...
 	if !c.EnableJavaTLSSupport || !c.EnableHTTPSMonitoring || !http.HTTPSSupported(c) {
-		return false
+		return nil
 	}
 
-	javaUSMAgentJarPath = filepath.Join(c.JavaDir, agentUSMJar)
-	jar, err := os.Open(javaUSMAgentJarPath)
-	if err != nil {
+	javaUSMAgentJarPath := filepath.Join(c.JavaDir, agentUSMJar)
+	if _, err := os.Stat(javaUSMAgentJarPath); err != nil {
```
Comment (val06) [inline pkg/network/usm/ebpf_javatls.go:131]: https://github.com/DataDog/datadog-agent/pull/14620/files#r1065824758

File: pkg/network/usm/ebpf_javatls.go [hunk 2]
```diff
@@ -36,195 +36,176 @@ const (
...
 			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				EBPFFuncName: "kprobe__do_vfs_ioctl",
+				EBPFFuncName: doVfsIoctlKprobeName,
 				UID:          probeUID,
 			},
 		})
 }
 
-func (p *JavaTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPair {
+func (p *javaTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPair {
```
Comment (val06) [inline pkg/network/usm/ebpf_javatls.go:195]: i think `p` is not used here and can be replace with `_` , does it pass the lint without this change btw?

File: pkg/network/usm/ebpf_javatls.go [hunk 3]
```diff
@@ -36,195 +36,176 @@ const (
...
+		{EBPFFuncName: handleConnectionByPeerKprobeName},
+		{EBPFFuncName: handleAsyncPayloadKprobeName},
 	}
 }
 
 // isJavaProcess checks if the given PID comm's name is java.
 // The method is much faster and efficient that using process.NewProcess(pid).Name().
-func isJavaProcess(pid int) bool {
-	filePath := filepath.Join(util.GetProcRoot(), strconv.Itoa(pid), "comm")
+func (p *javaTLSProgram) isJavaProcess(pid int) bool {
```
Comment (val06) [inline pkg/network/usm/ebpf_javatls.go:207]: why do you need to pass by pointer?

File: pkg/network/usm/ebpf_javatls.go [hunk 4]
```diff
@@ -246,16 +227,16 @@ func isJavaProcess(pid int) bool {
...
+// injectionAllowRegex has a higher priority
 //
 // # In case of only one regex (allow or block) is set, the regex will be evaluated as exclusive filter
 // /                 match  | not match
 // allowRegex only    true  | false
 // blockRegex only    false | true
-func isAttachmentAllowed(pid int) bool {
-	allowIsSet := javaAgentAllowRegex != nil
-	blockIsSet := javaAgentBlockRegex != nil
+func (p *javaTLSProgram) isAttachmentAllowed(pid int) bool {
```
Comment (val06) [inline pkg/network/usm/ebpf_javatls.go:237]: also here, why by pointer?

File: pkg/network/usm/ebpf_main.go [hunk 5]
```diff
@@ -144,19 +145,14 @@ func newEBPFProgram(c *config.Config, connectionProtocolMap, sockFD *ebpf.Map, b
...
 	if openSSLProg != nil {
 		subprograms = append(subprograms, openSSLProg)
 	}
 
-	var tailCalls []manager.TailCallRoute
-
-	if IsJavaSubprogramEnabled(c) {
-		tailCalls = append(tailCalls, GetJavaTlsTailCallRoutes()...)
-	}
-
```
Comment (val06) [inline pkg/network/usm/ebpf_main.go:159]: :+1:

**Review by val06 — CHANGES_REQUESTED** [review]
cred

File: pkg/network/usm/ebpf_javatls.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/usm/ebpf_javatls.go:131]: I wish that would have become a comment in the code from the first place

I'll revert and add comment

File: pkg/network/usm/ebpf_javatls.go (hunk 3 above)
Comment (guyarb) [reply pkg/network/usm/ebpf_javatls.go:207]: I need this function to be part of the program to access the members of the struct
the pointer receiver is not mandatory here, but a common practice in go say - if you have a single pointer receiver, for consistency, use pointer receiver everywhere

File: pkg/network/usm/ebpf_javatls.go (hunk 4 above)
Comment (guyarb) [reply pkg/network/usm/ebpf_javatls.go:237]: https://github.com/DataDog/datadog-agent/pull/18291#discussion_r1271271703

**Review by val06 — APPROVED** [review]

**Pull request merged**